piccolo/querystring.py

```diff
@@ -1,13 +1,19 @@
 from __future__ import annotations
 
-import datetime
 import typing as t
 from dataclasses import dataclass
+from datetime import datetime
+from importlib.util import find_spec
 from string import Formatter
 
 if t.TYPE_CHECKING:
     from piccolo.table import Table
 
+if find_spec("asyncpg"):
+    from asyncpg.pgproto.pgproto import UUID
+else:
+    from uuid import UUID
```

Comment:
Ideally it would check both of these, because for Postgres, a value could be a `uuid.UUID` or `asyncpg.pgproto.pgproto.UUID`.

We pass `uuid.UUID` into queries like INSERTS, but asyncpg returns `asyncpg.pgproto.pgproto.UUID` from SELECT. We're most likely to be dealing with `uuid.UUID`.

Reply:
`asyncpg` only gets installed if the `piccolo[postgres]` is installed, correct? I am not sure I have a good idea on how the case where it hasn't been installed other than moving the `find_spec("asyncpg")` into the conversion `if-elif-else`  scope. Is that what you would recommend?

Reply:
or something like

```python
if find_spec("asyncpg"):
    from asyncpg.pgproto.pgproto import UUID as apgUUID
else:
    from uuid import UUID
    apgUUID = UUID
 ```
Then do the appropriate typecheck on line 102

Reply:
Yes, that works.